### PR TITLE
gh: add options for "pager" and "prompt" to gh.nix

### PR DESCRIPTION
**Pull request by JackMoffat**
### Description

These two options are present in the config.yml file generated by
"gh auth login" with browser authentication. This change allows
gh.nix to generate a config.yml file that includes these options.

### Checklist

- [ ] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ ] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/gh.nix [hunk 1]
```diff
@@ -42,13 +42,32 @@ in {
...
+    };
+
+    prompt = mkOption {
+      type = types.enum [ "enabled" "disabled" ];
+      default = "enabled";
+      description = ''
+        When to interactively prompt.
+        This is a global config that cannot be overridden by hostname.
+        Supported values: enabled, disabled.
+        '';
```
Comment (sumnerevans) [inline modules/programs/gh.nix:62]:
```suggestion
      description = ''
        When to interactively prompt.
        This is a global config that cannot be overridden by hostname.
        Supported values: enabled, disabled.
      '';
```

File: modules/programs/gh.nix [hunk 2]
```diff
@@ -42,13 +42,32 @@ in {
...
+      type = types.str;
+      default = "";
+      description = ''
+        A pager program to send command output to, e.g. "less".
+        Set the value to "cat" to disable the pager.
+        '';
+    };
+
+    prompt = mkOption {
+      type = types.enum [ "enabled" "disabled" ];
```
Comment (sumnerevans) [inline modules/programs/gh.nix:56]: Maybe just make this a boolean?

File: modules/programs/gh.nix [hunk 3]
```diff
@@ -42,13 +42,32 @@ in {
...
     };
+
+    pager = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        A pager program to send command output to, e.g. "less".
+        Set the value to "cat" to disable the pager.
+        '';
+    };
```
Comment (sumnerevans) [inline modules/programs/gh.nix:53]:
```suggestion
    pager = mkOption {
      type = types.str;
      default = "";
      example = "less";
      description = ''
        The pager program to send command output to.
        Set the value to <literal>"cat"</literal> to disable the pager.
      '';
    };
```

File: modules/programs/gh.nix [hunk 4]
```diff
@@ -42,13 +42,32 @@ in {
...
+
+    prompt = mkOption {
+      type = types.enum [ "enabled" "disabled" ];
+      default = "enabled";
+      description = ''
+        When to interactively prompt.
```
Comment (sumnerevans) [inline modules/programs/gh.nix:59]:
```suggestion
        Whether to interactively prompt.
```

File: modules/programs/gh.nix [hunk 5]
```diff
@@ -42,13 +42,32 @@ in {
...
+        This is a global config that cannot be overridden by hostname.
+        Supported values: enabled, disabled.
```
Comment (sumnerevans) [inline modules/programs/gh.nix:61]: I'm pretty sure the options should already be rendered in the docs.

```suggestion
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Thanks for the contribution! Added some notes.

**Comment by berbiche** [comment]
Closing per my former comments: https://github.com/nix-community/home-manager/pull/2232#issuecomment-948925186

**Pull request closed without merging**